File: 백준/Silver/10815. 숫자 카드/숫자 카드.py
import sys
read_line = sys.stdin.readline

n = int(read_line())
cards = set(map(int, read_line().split()))

m = int(read_line())
query = list(map(int, read_line().split()))

print(" ".join(["1" if num in cards else "0" for num in query]))

File: 백준/Silver/10815. 숫자 카드/숫자 카드.java
import java.util.*;
import java.io.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        int n = Integer.parseInt(br.readLine());
        HashSet<Integer> cards = new HashSet<>();
        StringTokenizer st = new StringTokenizer(br.readLine());

        while (n-- > 0) {
            cards.add(Integer.parseInt(st.nextToken()));
        }

        int m = Integer.parseInt(br.readLine());
        st = new StringTokenizer(br.readLine());
        while (m-- > 0) {
            int num = Integer.parseInt(st.nextToken());
            sb.append(cards.contains(num) ? "1 " : "0 ");
        }

        br.close();
        System.out.println(sb.toString().trim());
    }
}